### PR TITLE
Correct rfid and weight timer

### PR DESCRIPTION
**Pull request by jw2476**
Simplifies the changes made in the Correct-RFID-&-Weight-timer branch

- [x] - Tested

### CODE REVIEW

**Comment by jw2476** [comment]
IT WORKS!!!

File: app/rfid.py [hunk 1]
```diff
@@ -11,26 +11,24 @@
...
-            ser.reset_input_buffer() #  clean buffer
-            ser.reset_output_buffer() # clean buffer
-            ser.write(b"sd2\r\n") # set mode of rfid
-            a = ser.read_until(size=19).decode(
-            "utf-8"
-            )  # 16 byte  + \r + \n somehow is 19 not 18
-            if len(a) > 15: # if read then return out of function
-                return a # return and break out of functions
+    def read(self):
+        ser.reset_input_buffer() #  clean buffer
```
Comment (sebbacon) [inline app/rfid.py]: The comment `# record for fixed time after pir reading` plus the fact there's a variable `rfid_record_time` set to 120 implies that something will record for 2 mins.

However, this function doesn't use that variable at all. Is that deliberate? If so, we should remove all reference to the variable here and in the calling function.

**Comment by jw2476** [comment]
It is, but my approach is a lot neater and has been tested.

**Comment by sebbacon** [comment]
Nope, I still don't understand :) 

As far as I can see, you've made a decision to completely remove the looping and just read a single set of 19 bytes?

If so, we need to remove all the reference to waiting and the variables that make it look like it waits.  Otherwise it's a trap for future users to get confused by,

**Comment by jw2476** [comment]
The read_until creates it's own loop that is run untill the timeout(120),
or the bytes to reach 19.

**Comment by sebbacon** [comment]
Ah, I see, it's in the setup of the serial interface.  All good!

**Pull request merged**